src/material/form-field/form-field.ts

```diff
@@ -83,13 +82,21 @@ const _MatFormFieldMixinBase: CanColorCtor & typeof MatFormFieldBase =
  */
 export type MatFormFieldAppearance = 'legacy' | 'standard' | 'fill' | 'outline';
 
+/** Possible values for the "floatLabel" form-field input. */
+export type FloatLabelType = 'always' | 'never' | 'auto';
+
 /**
  * Represents the default options for the form field that can be configured
  * using the `MAT_FORM_FIELD_DEFAULT_OPTIONS` injection token.
  */
 export interface MatFormFieldDefaultOptions {
   appearance?: MatFormFieldAppearance;
   hideRequiredMarker?: boolean;
+  /**
+   * Whether the label for form-fields should by default float `always`,
+   * `never`, or `auto` (only when necessary).
+   */
+  float?: FloatLabelType;
```

Comment:
Shouldn't this be `floatLabel` so it aligns with the input on `mat-form-field`?

Reply:
I don't know. I think it would make sense to align it with the input name since we do that for the other properties too. I'll change it.

Reply:
done. I also extracted the duplicated logic a bit. We should be able to reduce this logic in the MDC based implementation.